### PR TITLE
Experimental plugin system console UI

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Waiting on #7279 but should be ready for PM review.

Add some basic system console UI for plugins:
* Plugin section only shows up when `PluginSettings.Enable` is set to `true` in `config.json`
* Plugins can be uploaded, active plugins are listed, clicking remove will deactivate and delete a plugin
* Webapp plugins currently require users to manually refresh their browsers/apps to get new plugins or to have old plugins removed

#### Ticket Link
Part of https://mattermost.atlassian.net/browse/PLT-7274

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

**Comment by jwilander** [comment]
@jasonblais I've set the config so the plugins section should show up under `Integrations` in the system console

**Comment by jasonblais** [comment]
@jwilander Was there a sample plugin I could use to upload one, to test the UI?

I tried a few random ZIP files, which give an error (probably expected).

**Comment by jwilander** [comment]
I DMed you one earlier on pre-release

**Comment by jasonblais** [comment]
Oh right, sorry :)

**Comment by jasonblais** [comment]
@jwilander Can you help with the following changes:

1) Move plugins to E20 for now. 
2) Text changes to System Console page


TITLE 
`Plugins (Experimental)`

NOTE
`Plugins are experimental and not recommended for use in production.` // also remove the big "NOTE" from the top

SETTING TITLES
`Upload Plugin`
`Active Plugins`

HELP TEXT (for Upload Plugin)
`Upload a plugin for your Mattermost server. Adding or removing a webapp plugin requires users to refresh their browser or Desktop App before taking effect. See documentation to learn more.`
^where `documentation` links to https://about.mattermost.com/default-plugins

3) Error text changes

`Unable to activate extracted plugin. Plugin may already exist and be activated.` --> `Unable to upload the plugin. It may conflict with another plugin on your server.`

`Encountered error extracting plugin` --> `Encountered an error when extracting the plugin. Review your plugin file content and try again.`

----

@asaadmahmood Can you help with UI review and changes? I think there's three that we can do

4) Reduce whitespace below "Choose File" and "Upload" buttons
5) Align `No active plugins` text with `Active plugins` title
image

6) When a plugin is listed, perhaps move "ID" and "Description" to separate lines, similar to what we have in the personal access tokens screen?

image

**Comment by jwilander** [comment]
1. Making plugins EE only will come in a different PR
2. Will do
3. Will do

**Comment by jasonblais** [comment]
1. Sure thing.

**Comment by asaadmahmood** [comment]
Submitted a PR for 4, and 5.

**Comment by jwilander** [comment]
@jasonblais AM's changes have been merged and I've made the text updates

File: webapp/components/admin_console/plugin_settings/plugin_settings.jsx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+import PropTypes from 'prop-types';
+import {FormattedMessage, FormattedHTMLMessage} from 'react-intl';
+
+export default class PluginSettings extends React.Component {
+    static propTypes = {
+
+        /*
+         * The config
+         */
+        config: PropTypes.object.isRequired,
```
Comment (enahum) [inline webapp/components/admin_console/plugin_settings/plugin_settings.jsx]: is config pass in the `ownProps` object? cause I don't see it anywhere

**Comment by jasonblais** [comment]
Thanks @jwilander, it looks like the "Plugins" page isn't visible on the System Console sidebar. Tried on both an unlicensed and licensed server. Not sure if related to https://github.com/mattermost/platform/pull/7338/commits/bb008b23ea799559c0f521268858a57c8773901e#diff-93ca270a8430727d18f4a25af4340ae1R281

(I'm pretty sure the changes are all good, but would like to give it a quick test)

**Comment by jwilander** [comment]
I need to manually change it in config.json for the new spinmint

**Comment by jwilander** [comment]
Should be there now

**Comment by jasonblais** [comment]
@asaadmahmood Did you take a look at 6, or does this need a dev?

Current UI feels a bit awkward and won't extend easily to multiple plugins

image

**Comment by jasonblais** [comment]
Also, there's a few whitespace issues:

1) When trying to upload the same plugin twice
image

2) After choosing a file, before hitting the `Upload` button

image

**Pull request merged**